staging/src/k8s.io/client-go/applyconfigurations/core/v1/servicespec.go

```diff
@@ -65,17 +65,11 @@ func (b *ServiceSpecApplyConfiguration) WithPorts(values ...*ServicePortApplyCon
 	return b
 }
 
-// WithSelector puts the entries into the Selector field in the declarative configuration
-// and returns the receiver, so that objects can be build by chaining "With" function invocations.
-// If called multiple times, the entries provided by each call will be put on the Selector field,
-// overwriting an existing map entries in Selector field with the same key.
-func (b *ServiceSpecApplyConfiguration) WithSelector(entries map[string]string) *ServiceSpecApplyConfiguration {
-	if b.Selector == nil && len(entries) > 0 {
-		b.Selector = make(map[string]string, len(entries))
-	}
-	for k, v := range entries {
-		b.Selector[k] = v
-	}
```

Comment:
For some of these fields that arent []byte that are changing with this PR I am worried this API breaking change.

It is unlikely someone is calling With multiple times for []byte; but for map fields like this (and some other []string ones) I can imagine people who use `With...` multiple times getting bit by it longer merging/appending.

/cc @apelisse

Reply:
> For some of these fields that arent []byte that are changing with this PR I am worried this API breaking change.
> [...]
> I can imagine people who use With... multiple times getting bit by it longer merging/appending.

It is for sure a breaking change in behavior. I don't know if it's relevant considering the usage of this. @siyuanfoundation have you had a chance to search github for usage of this code in practice? And/or possibly even just that method specifically (`WithSelector`).

I think the new behavior for the atomic map is right though, I would expect people to set the map entirely since it's atomic, and piece by piece for non-atomic.

Reply:
there are usages of the `WithSelector` method like this one https://github.com/frankfanslc/liqo/blob/873a7609fdbecc98c8c5dcc013439e8f08dac934/pkg/virtualKubelet/forge/services.go#L38. 
Even though the new behavior should be the right way, it would probably break existing code. What would be the best path to fix this? Should we introduce new method names?
@apelisse

Reply:
We could have new method name and deprecate the existing ones, that's a reasonable option indeed.

Reply:
Curious what @jpbetz thinks of the name but here are some option:
- `ReplaceSelectors(selectors)`
- `WithNewSelectors(selectors)`
- `WithReplacedSelectors(selectors)`
- ...
WDYT?

Reply:
`WithNewSelectors(selectors)` SGTM